impeller/entity/entity_unittests.cc

```diff
@@ -2364,6 +2364,45 @@ APPLY_COLOR_FILTER_GRADIENT_TEST(Radial);
 APPLY_COLOR_FILTER_GRADIENT_TEST(Conical);
 APPLY_COLOR_FILTER_GRADIENT_TEST(Sweep);
 
+TEST_P(EntityTest, GiantStrokePathAllocation) {
+  PathBuilder builder{};
+  for (int i = 0; i < 10000; i++) {
+    builder.LineTo(Point(i, i));
+  }
+  Path path = builder.TakePath();
+  auto geom = Geometry::MakeStrokePath(path, /*stroke_width=*/10);
+
+  ContentContext content_context(GetContext(), nullptr);
+  Entity entity;
+
+  auto cmd_buffer = content_context.GetContext()->CreateCommandBuffer();
+
+  RenderTargetAllocator allocator(
+      content_context.GetContext()->GetResourceAllocator());
+
+  auto render_target =
+      allocator.CreateOffscreen(*content_context.GetContext(), {10, 10}, 1);
+  auto pass = cmd_buffer->CreateRenderPass(render_target);
+
+  GeometryResult result =
+      geom->GetPositionBuffer(content_context, entity, *pass);
+
+  // Validate the buffer data overflowed the small buffer
+  EXPECT_GT(result.vertex_buffer.vertex_count, kPointArenaSize);
+
+  // Validate that there are no uninitialized points near the gap.
+  Point* written_data = reinterpret_cast<Point*>(
+      (result.vertex_buffer.vertex_buffer.GetBuffer()->OnGetContents() +
+       result.vertex_buffer.vertex_buffer.GetRange().offset));
+
+  EXPECT_NE(*(written_data + kPointArenaSize - 2), Point(0, 0));
+  EXPECT_NE(*(written_data + kPointArenaSize - 1), Point(0, 0));
+  EXPECT_NE(*(written_data + kPointArenaSize), Point(0, 0));
+  EXPECT_NE(*(written_data + kPointArenaSize + 1), Point(0, 0));
+  EXPECT_NE(*(written_data + kPointArenaSize + 2), Point(0, 0));
+  EXPECT_NE(*(written_data + kPointArenaSize + 3), Point(0, 0));
```

Comment:
Since this is stroke tessellation the generated points are floating point values like 43.22132 ...

I could do a CLOSE_TO or something like that though. As long as they are in the ballpark

Reply:
Done